backend-update-and-delete_products/pages/api/products/[id].js

```diff
@@ -0,0 +1,30 @@
+import dbConnect from "../../../db/connect";
+import Product from "../../../db/models/Product";
+
+export default async function handler(request, response) {
+  await dbConnect();
+  const { id } = request.query;
+
+  if (request.method === "GET") {
+    const product = await Product.findById(id).populate("reviews");
+
+    if (!product) {
+      return response.status(404).json({ status: "Not Found" });
+    }
+
+    response.status(200).json(product);
+  }
+
+  if (request.method === "PUT") {
+    const updatedProduct = request.body;
+    await Product.findByIdAndUpdate(id, updatedProduct);
+    response.status(200).json({ message: "Product successfully updated" });
+    return;
+  }
+
+  if (request.method === "DELETE") {
+    await Product.findByIdAndDelete(id);
```

Comment:
what about the reviews connected to this product? They are not deleted yet.

Reply:
I am not sure what do you mean with reviews... When i delete the product, i can´t see reviews anymore.. 
If you mean the reviews and comments that i build in submit form, they are not in this PR and project.